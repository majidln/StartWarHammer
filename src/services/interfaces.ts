export interface Ingredient {
  carbs?: Number;
  fat?: Number;
  protein?: Number;
  fiber?: Number;
  calories?: Number;
}

export interface Nutrition {
  values: Ingredient;
  percentages: Ingredient;
}

export interface Tag {
  id: String;
  type: String;
  title: String;
}

export interface Instruction {
  title: String;
  steps: Array<String>;
  header?: Object;
  footer?: Object;
}

export interface Images {
  hz: String;
  vt: String;
  brightness: String;
}

export interface Recipe {
  id: string;
  slug: String;
  title: string;
  description: string;
  nutrition: Nutrition;
  time: Object;
  difficulty: Object;
  images: Images;
  tags: Array<Tag>;
  instructionSections: Array<Instruction>;
  rating: Number;
  isMembersOnly: Boolean;
  modifiedAt: String;
}

export interface RecipePage {
  listRecipes: Array<Recipe>;
  totalSize: Number;
  nextPage: Number;
}

export interface QueryResponse {
  data: RecipePage;
  loading: Boolean;
  fetchMore: Function;
}
