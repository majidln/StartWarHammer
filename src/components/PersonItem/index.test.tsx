import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, waitFor, fireEvent} from '@testing-library/react-native';
import {PersonItem} from './index';

import {Person} from '@services/interfaces';

const person: Person = {
  id: 'cj0nv9phmewhy0130he5s0k5r',
  name: 'Ackbar',
  height: 180,
  mass: 83,
  films: [
    {
      id: 'cj0nxmy5ega600114uiw05wle',
      title: 'Return of the Jedi',
    },
    {
      id: 'cj0nxmy69ga640114e3eb7bn2',
      title: 'The Force Awakens',
    },
  ],
  __typename: 'Person',
};

it('Person Item Snapshot renders correctly', () => {
  const tree = renderer.create(<PersonItem person={person} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Person Item render in list correctly', async () => {
  const {findByTestId} = render(<PersonItem person={person} />);

  await waitFor(() => findByTestId('personItem'));
});
