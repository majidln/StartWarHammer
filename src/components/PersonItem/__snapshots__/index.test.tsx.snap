// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Item Snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
  testID="personItem"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Height:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      180
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Mass:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      83
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Home World:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
        "marginVertical": 4,
        "width": "100%",
      }
    }
    testID="keyValueId"
  >
    <Text
      style={
        Object {
          "color": "#2c3e50",
          "fontSize": 18,
          "paddingRight": 4,
        }
      }
    >
      Play in:
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Return of the Jedi, The Force Awakens
    </Text>
  </View>
</View>
`;
